pkg/tsdb/azuremonitor/metrics/azuremonitor-datasource.go

```diff
@@ -74,33 +74,28 @@ func (e *AzureMonitorDatasource) buildQueries(queries []backend.DataQuery, dsInf
 		}
 
 		azJSONModel := queryJSONModel.AzureMonitor
+		// Legacy: If only MetricDefinition is set, use it as namespace
+		if azJSONModel.MetricDefinition != "" && azJSONModel.MetricNamespace == "" {
+			azJSONModel.MetricNamespace = azJSONModel.MetricDefinition
+		}
 
 		ub := urlBuilder{
 			ResourceURI: azJSONModel.ResourceURI,
-			// Legacy, used to reconstruct resource URI if it's not present
+			// Alternative, used to reconstruct resource URI if it's not present
 			DefaultSubscription: dsInfo.Settings.SubscriptionId,
 			Subscription:        queryJSONModel.Subscription,
 			ResourceGroup:       azJSONModel.ResourceGroup,
-			MetricDefinition:    azJSONModel.MetricDefinition,
+			MetricNamespace:     azJSONModel.MetricNamespace,
 			ResourceName:        azJSONModel.ResourceName,
 		}
 
 		azJSONModel.DimensionFilters = MigrateDimensionFilters(azJSONModel.DimensionFilters)
 		azureURL := ub.BuildMetricsURL()
 
-		resourceName := azJSONModel.ResourceName
-		if resourceName == "" {
-			resourceName = extractResourceNameFromMetricsURL(azureURL)
+		if azJSONModel.ResourceName == "" {
+			azJSONModel.ResourceName = extractResourceNameFromMetricsURL(azureURL)
 		}
 
-		urlComponents := map[string]string{}
-		urlComponents["resourceURI"] = azJSONModel.ResourceURI
-		// Legacy fields used for constructing a deep link to display the query in Azure Portal.
-		urlComponents["subscription"] = queryJSONModel.Subscription
-		urlComponents["resourceGroup"] = azJSONModel.ResourceGroup
-		urlComponents["metricDefinition"] = azJSONModel.MetricDefinition
-		urlComponents["resourceName"] = resourceName
```

Comment:
removed this map since it was untyped (and replaced with typed properties for the ones used)